### PR TITLE
[py]qtwebengine: meta.platforms; use in qutebrowser, calibre

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

This four-patch series does the following:

1. `qtwebengine`: implement `meta.platforms` correctly, based on the    current state as of qtwebengine 5.15.8.  Previously, meta.platforms    was simply set to platforms.unix, which is certainly not correct.    In particular, powerpc64le is not supported right now (but see below)

2. `pyqtwebengine`: do not claim enablement for a platform which is    unsupported by `qtwebengine`.

3. `qutebrowser`: do not claim enablement on platforms where the    underlying backend (`pyqtwebkit` or `pyqtwebengine`) is not    enabled.  This is straightforward; if the chosen backend does not    work then `qutebrowser` cannot do much.

4. `calibre`: on platforms where `qtwebengine` is not enabled, simply    omit `qtwebengine` rather than failing the build.  Most of calibre's    functionality does not require a web browser, so disabling calibre    simply because `qtwebengine` is unavailable is overaggressive.

As of this moment `qtwebengine` is not supported on powerpc64le. However that is likely to change soon; the upstream chromium has a full javascript JIT for powerpc64le (which is already used in `nodejs` on powerpc64le), and gentoo is shipping a usable qtwebengine on ppc64:

  https://packages.gentoo.org/packages/dev-qt/qtwebengine

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [X] powerpc64le-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/qutebrowser/default.nix [hunk 1]
```diff
@@ -132,5 +132,6 @@ in mkDerivationWith python3Packages.buildPythonApplication rec {
     description = "Keyboard-focused browser with a minimal GUI";
     license     = licenses.gpl3Plus;
     maintainers = with maintainers; [ jagajaga rnhmjoj ebzzry dotlambda ];
+    platforms   = backendPackage.meta.platforms;
```
[the remark this answers is not in the document]
Comment (a-m-joseph) [reply pkgs/applications/networking/browsers/qutebrowser/default.nix]: [417f51d](https://github.com/NixOS/nixpkgs/pull/170196/commits/417f51d50edac2152733494bd87bc591728e8c43)

File: pkgs/development/libraries/qt-5/modules/qtwebengine.nix [hunk 2]
```diff
@@ -240,7 +240,21 @@ qtModule {
   meta = with lib; {
     description = "A web engine based on the Chromium web browser";
     maintainers = with maintainers; [ matthewbauer ];
-    platforms = platforms.unix;
+
+    # qtwebengine-5.15.8: "QtWebEngine can only be built for x86, x86-64, ARM, Aarch64, and MIPSel architectures."
```
Comment (AndersonTorres) [inline pkgs/development/libraries/qt-5/modules/qtwebengine.nix:244]: Split the comment in the classic 80-char limit.

Comment (a-m-joseph) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix:244]: Done.

File: pkgs/applications/misc/calibre/default.nix [hunk 3]
```diff
@@ -114,6 +113,8 @@ mkDerivation rec {
       pycryptodome
       # the following are distributed with calibre, but we use upstream instead
       odfpy
+    ] ++ lib.optionals (lib.lists.any (p: p == stdenv.hostPlatform.system) pyqtwebengine.meta.platforms) [
```
[the remark this answers is not in the document]
Comment (a-m-joseph) [reply pkgs/applications/misc/calibre/default.nix:116]: Done.

File: pkgs/development/libraries/qt-5/modules/qtwebengine.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix]: FYI We don't care about 80 chars. 120 or 150 are also fine.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/libraries/qt-5/modules/qtwebengine.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix]: There is/was a slide bar in the web version of github.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Everything is OK. Now rebase/squash the commits, cleaning them up (no need for those meta-comments).

**Comment by a-m-joseph** [comment]
Squashed back to four commits.  Let me know if you'd like it all squashed to a single commit.

**Review by AndersonTorres — APPROVED** [review]
LGTM, LGTBorg.

**Pull request merged**